{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Stage 1: Search a template"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def code2pirseq(code):\n",
    "    from Bio import ExPASy, SeqIO\n",
    "    spraw = ExPASy.get_sprot_raw(code)\n",
    "    spinfo = SeqIO.read(spraw, \"swiss\")\n",
    "    PDBs = [x[4:] for x in spinfo.dbxrefs if x.startswith('PDB:')]\n",
    "    #print len(PDBs), PDBs\n",
    "    \n",
    "    # Name of the protein (a name without a white space is preferred)\n",
    "    name     = str(spinfo.name)\n",
    "    sequence = str(spinfo.seq)\n",
    "    outname  = \"%s.ali\" % name\n",
    "    with open(outname, \"w\") as f:\n",
    "        f.write(\">P1;%s\\n\" % name)\n",
    "        f.write(\"sequence:%s: %d:A:%d: A:::0.00: 0.00\\n\" % (name, 1, len(sequence)) )\n",
    "        f.write(\"%s*\\n\" % sequence)\n",
    "    return name, outname, sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GTR1_RAT.ali\n"
     ]
    }
   ],
   "source": [
    "# Verificar que obtener un archivo de nuestra secuencia de inetres en formato PIR\n",
    "code = \"P11167\"\n",
    "name, outfile, sequence = code2pirseq(code)\n",
    "print(outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-12 16:22:23--  https://salilab.org/modeller/downloads/pdb_95.pir.gz\n",
      "Resolviendo salilab.org (salilab.org)... 169.230.79.19\n",
      "Conectando con salilab.org (salilab.org)[169.230.79.19]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: 17047044 (16M) [application/x-gzip]\n",
      "Grabando a: «pdb_95.pir.gz»\n",
      "\n",
      "pdb_95.pir.gz       100%[===================>]  16.26M  2.62MB/s    en 8.8s    \n",
      "\n",
      "2022-04-12 16:22:33 (1.84 MB/s) - «pdb_95.pir.gz» guardado [17047044/17047044]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://salilab.org/modeller/downloads/pdb_95.pir.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Para obetenr la base de datos de PDB no redundantes desde Modeller\n",
    "#!wget https://salilab.org/modeller/downloads/pdb_95.pir.gz\n",
    "\n",
    "def findTemplate(spcode):\n",
    "    name, outname, sequence = code2pirseq(spcode)\n",
    "    pdb_95_pir = \"pdb_95.pir.gz\"\n",
    "    seqQuery   = outname\n",
    "    \n",
    "    from modeller import environ, log, alignment, sequence_db\n",
    "    \n",
    "    log.verbose()\n",
    "    env = environ()\n",
    "    \n",
    "    #-- Prepare the input files\n",
    "    \n",
    "    #-- Read in the sequence database\n",
    "    sdb = sequence_db(env)\n",
    "    sdb.read(seq_database_file=pdb_95_pir, seq_database_format='PIR',\n",
    "             chains_list='ALL', minmax_db_seq_len=(30, 4000), clean_sequences=True)\n",
    "    \n",
    "    #-- Write the sequence database in binary form\n",
    "    sdb.write(seq_database_file='pdb_95.bin', seq_database_format='BINARY',\n",
    "              chains_list='ALL')\n",
    "    \n",
    "    #-- Now, read in the binary database\n",
    "    sdb.read(seq_database_file='pdb_95.bin', seq_database_format='BINARY',\n",
    "             chains_list='ALL')\n",
    "    \n",
    "    #-- Read in the target sequence/alignment\n",
    "    aln = alignment(env)\n",
    "    aln.append(file=seqQuery, alignment_format='PIR', align_codes='ALL')\n",
    "    \n",
    "    #-- Convert the input sequence/alignment into\n",
    "    #   profile format\n",
    "    prf = aln.to_profile()\n",
    "    \n",
    "    #-- Scan sequence database to pick up homologous sequences\n",
    "    prf.build(sdb, matrix_offset=-450, rr_file='${LIB}/blosum62.sim.mat',\n",
    "              gap_penalties_1d=(-500, -50), n_prof_iterations=1,\n",
    "              check_profile=False, max_aln_evalue=0.01)\n",
    "    \n",
    "    #-- Write out the profile in text format\n",
    "    prf.write(file='%s_profile.prf' % name, profile_format='TEXT')\n",
    "    \n",
    "    #-- Convert the profile back to alignment format\n",
    "    aln = prf.to_alignment()\n",
    "    \n",
    "    #-- Write out the alignment file\n",
    "    aln.write(file='%s_profile.ali' % name, alignment_format='PIR')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       445980     435.527     0.425\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       446508     436.043     0.426\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       495108     483.504     0.472\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       508408     496.492     0.485\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "sequence_d_W> The class 'sequence_db' is deprecated; use 'SequenceDB' instead\n",
      "Uncompressing compressed file with system command:\n",
      "gzip -d -c pdb_95.pir.gz > /var/folders/3w/2f_6xc8s5kj8yydws3xm1psh0000gn/T/pdb_95.pir.B2VHK1\n",
      "openf___224_> Open           pdb_95.pir.gz\n",
      "readlinef__W> File: pdb_95.pir.gz, Line: 6404\n",
      "              Modeller will only read the first 4096 characters of this line.\n",
      "\n",
      "readlinef__W> File: pdb_95.pir.gz, Line: 6479\n",
      "              Modeller will only read the first 4096 characters of this line.\n",
      "\n",
      "readlinef__W> File: pdb_95.pir.gz, Line: 8459\n",
      "              Modeller will only read the first 4096 characters of this line.\n",
      "\n",
      "readlinef__W> File: pdb_95.pir.gz, Line: 10488\n",
      "              Modeller will only read the first 4096 characters of this line.\n",
      "\n",
      "readlinef__W> File: pdb_95.pir.gz, Line: 32904\n",
      "              Modeller will only read the first 4096 characters of this line.\n",
      "\n",
      "readlinef__W> File: pdb_95.pir.gz, Line: 32910\n",
      "              No more warnings will be printed for truncated lines in this file.\n",
      "\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:       790134     771.615     0.754\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:      4516352    4410.500     4.307\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      4518594    4412.689     4.309\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      4520044    4414.105     4.311\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      4522944    4416.938     4.313\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      4528744    4422.602     4.319\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:     18590143   18154.438    17.729\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     18601743   18165.766    17.740\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     18624943   18188.422    17.762\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     18671343   18233.734    17.806\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     18764143   18324.359    17.895\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:     25794843   25190.277    24.600\n",
      "\n",
      "SEQ_DATABASE_FILE         : pdb_95.pir.gz\n",
      "SEQ_DATABASE_FORMAT       : PIR\n",
      "CHAINS_LIST               : ALL\n",
      "CLEAN_SEQUENCES           :            T\n",
      "MINMAX_DB_SEQ_LEN         :           30        4000\n",
      "Number of sequences       :        70307\n",
      "Number of residues        :     18064742\n",
      "Length of longest sequence:         3725\n",
      "\n",
      "\n",
      "SEQ_DATABASE_FILE         : pdb_95.bin\n",
      "SEQ_DATABASE_FORMAT       : BINARY\n",
      "SEARCH_CHAINS_LIST        : ALL\n",
      "Number of sequences       :        70307\n",
      "Number of residues        :     18064742\n",
      "Length of longest sequence:         3725\n",
      "\n",
      "\n",
      "SEQ_DATABASE_FILE         : pdb_95.bin\n",
      "SEQ_DATABASE_FORMAT       : BINARY\n",
      "CHAINS_LIST               : ALL\n",
      "CLEAN_SEQUENCES           :            T\n",
      "MINMAX_DB_SEQ_LEN         :            0      999999\n",
      "Number of sequences       :        70307\n",
      "Number of residues        :     18064742\n",
      "Length of longest sequence:         3725\n",
      "\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "openf___224_> Open           GTR1_RAT.ali\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       792454     773.881     0.756\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       793904     775.297     0.757\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       796804     778.129     0.760\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       802604     783.793     0.765\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:       814204     795.121     0.776\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       816168     797.039     0.778\n",
      "\n",
      "Read the alignment from file       : GTR1_RAT.ali\n",
      "\n",
      "Total number of alignment positions:   492\n",
      "\n",
      "  #  Code        #_Res #_Segm PDB_code    Name\n",
      "-------------------------------------------------------------------------------\n",
      "  1   GTR1_RAT     492      1    GTR1_RAT\n",
      "\n",
      "Dynamically allocated memory at     amaxprofile [B,KiB,MiB]:       818469     799.286     0.781\n",
      "openf___224_> Open           ${LIB}/blosum62.sim.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:      1141139    1114.394     1.088\n",
      "profile_iteration_> processing sequence:       1    492      1     0.0100000     0.0100000     0.0100000     1\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:      1146296    1119.430     1.093\n",
      "profile_iteration_> processing sequence:    7032    492      1     2.3599999     0.0003356     0.0000013  2980\n",
      "profile_iteration_> processing sequence:   14063    492      1     4.5700002     0.0003250     0.0000013  3077\n",
      "profile_iteration_> processing sequence:   21094    492      1     6.7300000     0.0003190     0.0000012  3134\n",
      "profile_iteration_> processing sequence:   28125    492      1     8.9899998     0.0003196     0.0000013  3128\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:      1147641    1120.743     1.094\n",
      "profile_iteration_> processing sequence:   35156    492      1    11.3500004     0.0003228     0.0000013  3097\n",
      "profile_iteration_> processing sequence:   42187    492      1    13.6599998     0.0003238     0.0000013  3088\n",
      "profile_iteration_> processing sequence:   49218    492      1    15.9099998     0.0003233     0.0000013  3094\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:      1149769    1122.821     1.097\n",
      "profile_iteration_> processing sequence:   56249    492      1    18.1399994     0.0003225     0.0000013  3101\n",
      "profile_iteration_> processing sequence:   63280    492      1    20.2900009     0.0003206     0.0000013  3119\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:      1151723    1124.729     1.098\n",
      "\n",
      "Dynamically allocated memory at amaxsequence_db [B,KiB,MiB]:      1153891    1126.847     1.100\n",
      " AVERAGE RESIDUAL STANDARD ERROR:    32.32372\n",
      " AVERAGE RESIDUAL STANDARD DEVIATION:    237.2840\n",
      "profile_660_> Quality of the fit between observed/expected score distributions.\n",
      "              [Iteration, Chi2, nbins, KS-Stat, nscores ] =        1,        0.0169,      121,        0.0052,    70307\n",
      "\n",
      "HISTOGRAM OF Z-SCORES:\n",
      "\n",
      "Z:         1  -3.95000   0.00000   0.00000\n",
      "Z:         1  -3.85000   0.00000   0.00000\n",
      "Z:         1  -3.75000   0.00000   0.00000\n",
      "Z:         1  -3.65000   0.00000   0.00000\n",
      "Z:         1  -3.55000   0.00000   0.00000\n",
      "Z:         1  -3.45000   0.00000   0.00000\n",
      "Z:         1  -3.35000   0.00000   0.00000\n",
      "Z:         1  -3.25000   0.00000   0.00000\n",
      "Z:         1  -3.15000   0.00000   0.00000\n",
      "Z:         1  -3.05000   0.00000   0.00000\n",
      "Z:         1  -2.95000   0.00000   0.00000\n",
      "Z:         1  -2.85000   0.00000   0.00000\n",
      "Z:         1  -2.75000   0.00000   0.00000\n",
      "Z:         1  -2.65000   0.00014   0.00000\n",
      "Z:         1  -2.55000   0.00028   0.00001\n",
      "Z:         1  -2.45000   0.00014   0.00004\n",
      "Z:         1  -2.35000   0.00028   0.00016\n",
      "Z:         1  -2.25000   0.00171   0.00055\n",
      "Z:         1  -2.15000   0.00299   0.00163\n",
      "Z:         1  -2.05000   0.00512   0.00416\n",
      "Z:         1  -1.95000   0.00925   0.00933\n",
      "Z:         1  -1.85000   0.02034   0.01872\n",
      "Z:         1  -1.75000   0.04239   0.03400\n",
      "Z:         1  -1.65000   0.05533   0.05658\n",
      "Z:         1  -1.55000   0.09387   0.08721\n",
      "Z:         1  -1.45000   0.11236   0.12565\n",
      "Z:         1  -1.35000   0.16186   0.17059\n",
      "Z:         1  -1.25000   0.20396   0.21981\n",
      "Z:         1  -1.15000   0.28532   0.27052\n",
      "Z:         1  -1.05000   0.29400   0.31973\n",
      "Z:         1  -0.95000   0.36398   0.36469\n",
      "Z:         1  -0.85000   0.40224   0.40316\n",
      "Z:         1  -0.75000   0.44192   0.43360\n",
      "Z:         1  -0.65000   0.45813   0.45519\n",
      "Z:         1  -0.55000   0.47349   0.46779\n",
      "Z:         1  -0.45000   0.49426   0.47182\n",
      "Z:         1  -0.35000   0.45472   0.46811\n",
      "Z:         1  -0.25000   0.43339   0.45775\n",
      "Z:         1  -0.15000   0.43424   0.44196\n",
      "Z:         1  -0.05000   0.42869   0.42194\n",
      "Z:         1   0.05000   0.39569   0.39889\n",
      "Z:         1   0.15000   0.37322   0.37383\n",
      "Z:         1   0.25000   0.36227   0.34769\n",
      "Z:         1   0.35000   0.34989   0.32121\n",
      "Z:         1   0.45000   0.28731   0.29500\n",
      "Z:         1   0.55000   0.26071   0.26953\n",
      "Z:         1   0.65000   0.23682   0.24513\n",
      "Z:         1   0.75000   0.21406   0.22205\n",
      "Z:         1   0.85000   0.20055   0.20043\n",
      "Z:         1   0.95000   0.17608   0.18036\n",
      "Z:         1   1.05000   0.16712   0.16185\n",
      "Z:         1   1.15000   0.14607   0.14489\n",
      "Z:         1   1.25000   0.13185   0.12944\n",
      "Z:         1   1.35000   0.11123   0.11542\n",
      "Z:         1   1.45000   0.10440   0.10275\n",
      "Z:         1   1.55000   0.08363   0.09133\n",
      "Z:         1   1.65000   0.08434   0.08109\n",
      "Z:         1   1.75000   0.07169   0.07191\n",
      "Z:         1   1.85000   0.05917   0.06371\n",
      "Z:         1   1.95000   0.05206   0.05639\n",
      "Z:         1   2.05000   0.05192   0.04988\n",
      "Z:         1   2.15000   0.04722   0.04409\n",
      "Z:         1   2.25000   0.03883   0.03895\n",
      "Z:         1   2.35000   0.03783   0.03439\n",
      "Z:         1   2.45000   0.03101   0.03035\n",
      "Z:         1   2.55000   0.02887   0.02678\n",
      "Z:         1   2.65000   0.02276   0.02361\n",
      "Z:         1   2.75000   0.02162   0.02082\n",
      "Z:         1   2.85000   0.01934   0.01835\n",
      "Z:         1   2.95000   0.01806   0.01617\n",
      "Z:         1   3.05000   0.01579   0.01424\n",
      "Z:         1   3.15000   0.01223   0.01255\n",
      "Z:         1   3.25000   0.01181   0.01105\n",
      "Z:         1   3.35000   0.00839   0.00973\n",
      "Z:         1   3.45000   0.00996   0.00857\n",
      "Z:         1   3.55000   0.00896   0.00754\n",
      "Z:         1   3.65000   0.00697   0.00664\n",
      "Z:         1   3.75000   0.00526   0.00584\n",
      "Z:         1   3.85000   0.00797   0.00514\n",
      "Z:         1   3.95000   0.00555   0.00453\n",
      "Z:         1   4.05000   0.00398   0.00398\n",
      "Z:         1   4.15000   0.00427   0.00350\n",
      "Z:         1   4.25000   0.00370   0.00308\n",
      "Z:         1   4.35000   0.00284   0.00271\n",
      "Z:         1   4.45000   0.00242   0.00239\n",
      "Z:         1   4.55000   0.00242   0.00210\n",
      "Z:         1   4.65000   0.00313   0.00185\n",
      "Z:         1   4.75000   0.00327   0.00163\n",
      "Z:         1   4.85000   0.00299   0.00143\n",
      "Z:         1   4.95000   0.00171   0.00126\n",
      "Z:         1   5.05000   0.00199   0.00111\n",
      "Z:         1   5.15000   0.00128   0.00097\n",
      "Z:         1   5.25000   0.00114   0.00086\n",
      "Z:         1   5.35000   0.00114   0.00075\n",
      "Z:         1   5.45000   0.00057   0.00066\n",
      "Z:         1   5.55000   0.00085   0.00058\n",
      "Z:         1   5.65000   0.00014   0.00051\n",
      "Z:         1   5.75000   0.00128   0.00045\n",
      "Z:         1   5.85000   0.00071   0.00040\n",
      "Z:         1   5.95000   0.00028   0.00035\n",
      "Z:         1   6.05000   0.00028   0.00031\n",
      "Z:         1   6.15000   0.00043   0.00027\n",
      "Z:         1   6.25000   0.00085   0.00024\n",
      "Z:         1   6.35000   0.00043   0.00021\n",
      "Z:         1   6.45000   0.00028   0.00018\n",
      "Z:         1   6.55000   0.00043   0.00016\n",
      "Z:         1   6.65000   0.00014   0.00014\n",
      "Z:         1   6.75000   0.00000   0.00013\n",
      "Z:         1   6.85000   0.00000   0.00011\n",
      "Z:         1   6.95000   0.00014   0.00010\n",
      "Z:         1   7.05000   0.00028   0.00009\n",
      "Z:         1   7.15000   0.00000   0.00007\n",
      "Z:         1   7.25000   0.00014   0.00007\n",
      "Z:         1   7.35000   0.00043   0.00006\n",
      "Z:         1   7.45000   0.00014   0.00005\n",
      "Z:         1   7.55000   0.00028   0.00004\n",
      "Z:         1   7.65000   0.00014   0.00004\n",
      "Z:         1   7.75000   0.00000   0.00003\n",
      "Z:         1   7.85000   0.00014   0.00003\n",
      "Z:         1   7.95000   0.00000   0.00003\n",
      "Z:         1   8.05000   0.00000   0.00002\n",
      "\n",
      "HITS FOUND IN ITERATION:     1\n",
      "\n",
      "\n",
      "Dynamically allocated memory at     amaxprofile [B,KiB,MiB]:      1564491    1527.823     1.492\n",
      "> 4zw9A                      1    4543   81400     470     492   67.60     0.0           2   462     5   467     3   465\n",
      "> 4ja3A                      1    7883   21350     426     492   30.94     0.0           3   392    19   454    10   426\n",
      "> 6m20A                      1   17115   21950     476     492   28.73     0.0           4   430    19   457    11   459\n",
      "> 6h7dA                      1   19898   21650     487     492   26.33     0.0           5   449     8   469     1   471\n",
      "> 4qiqA                      1   37407   21450     439     492   30.77     0.0           6   398    19   459    10   438\n",
      "> 4ybqA                      1   38809   48050     452     492   45.43     0.0           7   435     7   463     5   442\n",
      "> 4yb9D                      1   38851   47350     437     492   43.56     0.0           8   425    23   454    11   437\n",
      "> 5c65A                      1   46458   79300     457     492   68.43     0.0           9   452     7   467     1   453\n",
      "> 4ldsA                      1   50907   27200     421     492   32.85     0.0          10   411    21   457     8   421\n",
      "> 6thaA                      1   51591  111350     448     492   97.54     0.0          11   448     8   455     1   448\n",
      "> 4gc0A                      1   56906   24650     475     492   30.57     0.0          12   425    19   463    13   470\n",
      "> 6rw3A                      1   65118   21400     437     492   29.48     0.0          13   415    19   457    12   435\n",
      "> 6rw3B                      1   65119   20850     439     492   29.91     0.0          14   413    19   456    12   439\n",
      "\n",
      "profile_iteration__> Weight Matrix           [sec]:     0.010000\n",
      "profile_iteration__> Score Collection        [sec]:    22.580000\n",
      "profile_iteration__> Score Collection/Seq    [sec]:     0.000321\n",
      "profile_iteration__> DP Steps per sec    [ali/sec]:  3137.304688\n",
      "profile_iteration__> Statistics              [sec]:     0.050000\n",
      "profile_iteration__> Add to profile          [sec]:     0.030000\n",
      "profile_iteration__> Overall performance    [/seq]:     0.000324\n",
      "profile_iteration__> Overall performance [seq/sec]:  3090.417480\n",
      "openf___224_> Open           GTR1_RAT_profile.prf\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1309237    1278.552     1.249\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1311201    1280.470     1.250\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1313045    1282.271     1.252\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1314609    1283.798     1.254\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1316325    1285.474     1.255\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1318117    1287.224     1.257\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1319705    1288.774     1.259\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1321441    1290.470     1.260\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1323137    1292.126     1.262\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1324941    1293.888     1.264\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1326581    1295.489     1.265\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1328369    1297.235     1.267\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1330065    1298.892     1.268\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1331721    1300.509     1.270\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1333369    1302.118     1.272\n",
      "openf___224_> Open           GTR1_RAT_profile.ali\n"
     ]
    }
   ],
   "source": [
    "findTemplate(spcode=\"P11167\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-04-12 16:25:53--  https://files.rcsb.org/download/4zw9.pdb\n",
      "Resolviendo files.rcsb.org (files.rcsb.org)... 128.6.158.70\n",
      "Conectando con files.rcsb.org (files.rcsb.org)[128.6.158.70]:443... conectado.\n",
      "Petición HTTP enviada, esperando respuesta... 200 OK\n",
      "Longitud: no especificado [application/octet-stream]\n",
      "Grabando a: «4zw9.pdb»\n",
      "\n",
      "4zw9.pdb                [     <=>            ] 654.57K   644KB/s    en 1.0s    \n",
      "\n",
      "2022-04-12 16:25:54 (644 KB/s) - «4zw9.pdb» guardado [670275]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://files.rcsb.org/download/4zw9.pdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       445980     435.527     0.425\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       446508     436.043     0.426\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       495108     483.504     0.472\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:       508408     496.492     0.485\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "model______W> The class 'model' is deprecated; use 'Model' instead\n",
      "openf___224_> Open           4ZW9.pdb\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       520618     508.416     0.497\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       520694     508.490     0.497\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       520865     508.657     0.497\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       521715     509.487     0.498\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       522973     510.716     0.499\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       524877     512.575     0.501\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       524917     512.614     0.501\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       525007     512.702     0.501\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       527863     515.491     0.503\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       527923     515.550     0.503\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       528058     515.682     0.504\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       532342     519.865     0.508\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       532430     519.951     0.508\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       532628     520.145     0.508\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       539054     526.420     0.514\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       539186     526.549     0.514\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       539483     526.839     0.514\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       549105     536.235     0.524\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       549305     536.431     0.524\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       549755     536.870     0.524\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       564205     550.981     0.538\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       564505     551.274     0.538\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       565180     551.934     0.539\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       586838     573.084     0.560\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       587286     573.521     0.560\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       588294     574.506     0.561\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       620798     606.248     0.592\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:       621470     606.904     0.593\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       622982     608.381     0.594\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:       671738     655.994     0.641\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      1019578     995.682     0.972\n",
      "read_mo_297_> Segments, residues, atoms:        1      470     3607\n",
      "read_mo_298_> Segment:        1  1:A  470:A     3607\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1033179    1008.964     0.985\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      1302939    1272.401     1.243\n",
      "openf___224_> Open           GTR1_RAT.ali\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1451677    1417.653     1.384\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1596437    1559.021     1.522\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      1598401    1560.938     1.524\n",
      "\n",
      "Read the alignment from file       : GTR1_RAT.ali\n",
      "\n",
      "Total number of alignment positions:   492\n",
      "\n",
      "  #  Code        #_Res #_Segm PDB_code    Name\n",
      "-------------------------------------------------------------------------------\n",
      "  1       4ZW9     470      1    4ZW9.pdb MOL_ID  1; MOLECULE  SOLUTE CARRIER FAMILY 2, FACILITATED GLUCOSE TRAN MEMBER 3;\n",
      "  2   GTR1_RAT     492      1    GTR1_RAT\n",
      "align2d_276_> 'align_block' changed to 1.\n",
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  N -->  N\n",
      "              This message is written only for the first such atom.\n",
      "openf___224_> Open           $(LIB)/as1.sim.mat\n",
      "rdrrwgh_268_> Number of residue types:       20\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :           40\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :          492\n",
      "  Score                    :  358142.7500\n",
      "openf___224_> Open           GTR1_RAT-4ZW9.ali\n",
      "openf___224_> Open           GTR1_RAT-4ZW9.pap\n"
     ]
    }
   ],
   "source": [
    "from modeller import *\n",
    "#Alineamiento de la secuencia y el templado.\n",
    "env = environ()\n",
    "aln = alignment(env)\n",
    "mdl = model(env, file='4ZW9', model_segment=('FIRST:A','LAST:A'))\n",
    "aln.append_model(mdl, align_codes='4ZW9', atom_files='4ZW9.pdb')\n",
    "aln.append(file='GTR1_RAT.ali', align_codes='GTR1_RAT')\n",
    "aln.align2d(max_gap_length=40)\n",
    "aln.write(file='GTR1_RAT-4ZW9.ali', alignment_format='PIR')\n",
    "aln.write(file='GTR1_RAT-4ZW9.pap', alignment_format='PAP')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:      1789967    1748.015     1.707\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:      1790495    1748.530     1.708\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:      1839095    1795.991     1.754\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:      1852395    1808.979     1.767\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "automodel__W> The class 'automodel' is deprecated; use 'AutoModel' instead\n",
      "openf___224_> Open           GTR1_RAT-4ZW9.ali\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1864168    1820.477     1.778\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1865618    1821.893     1.779\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1868518    1824.725     1.782\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1874318    1830.389     1.787\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      1885918    1841.717     1.799\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:      2126726    2076.881     2.028\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2128602    2078.713     2.030\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2130566    2080.631     2.032\n",
      "\n",
      "Read the alignment from file       : GTR1_RAT-4ZW9.ali\n",
      "\n",
      "Total number of alignment positions:   492\n",
      "\n",
      "  #  Code        #_Res #_Segm PDB_code    Name\n",
      "-------------------------------------------------------------------------------\n",
      "  1       4ZW9     470      1    4ZW9.pdb MOL_ID  1; MOLECULE  SOLUTE CARRIER FAMILY 2, FACILITATED GLUCOSE TRAN MEMBER 3;\n",
      "  2   GTR1_RAT     492      1    GTR1_RAT\n",
      "check_a_343_> >> BEGINNING OF COMMAND\n",
      "openf___224_> Open           4ZW9.pdb\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2273291    2220.011     2.168\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2273291    2220.011     2.168\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2273462    2220.178     2.168\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2274312    2221.008     2.169\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2275570    2222.236     2.170\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2277474    2224.096     2.172\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2277474    2224.096     2.172\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2277564    2224.184     2.172\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2280420    2226.973     2.175\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2280420    2226.973     2.175\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2280555    2227.104     2.175\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2284839    2231.288     2.179\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2284839    2231.288     2.179\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2285037    2231.481     2.179\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2291463    2237.757     2.185\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2291463    2237.757     2.185\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2291760    2238.047     2.186\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2301382    2247.443     2.195\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2301382    2247.443     2.195\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2301832    2247.883     2.195\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2316282    2261.994     2.209\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2316282    2261.994     2.209\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2316957    2262.653     2.210\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2338615    2283.804     2.230\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2338615    2283.804     2.230\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2339623    2284.788     2.231\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2372127    2316.530     2.262\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:      2372127    2316.530     2.262\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2373639    2318.007     2.264\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:      2422395    2365.620     2.310\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      2422080    2365.312     2.310\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "check_a_344_> << END OF COMMAND\n",
      "openf___224_> Open           ${LIB}/top_heav.lib\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "openf___224_> Open           ${LIB}/par.lib\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2561332    2501.301     2.443\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2562932    2502.863     2.444\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2566912    2506.750     2.448\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2568912    2508.703     2.450\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2571912    2511.633     2.453\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2576412    2516.027     2.457\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2581984    2521.469     2.462\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2584784    2524.203     2.465\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2588984    2528.305     2.469\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2595284    2534.457     2.475\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2598056    2537.164     2.478\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:      2599456    2538.531     2.479\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                              227     561     661     112       0\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      2676543    2613.812     2.553\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3051001    2979.493     2.910\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3108465    3035.610     2.964\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3802   3891      0         0      1646\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3271949    3195.263     3.120\n",
      "\n",
      "patch_______> segment topology patched using RTF:  1 ;  MET  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3802   3891   5278      6261      1646\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      3431005    3350.591     3.272\n",
      "patch_______> segment topology patched using RTF:  492 ;  VAL  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3803   3892   5280      6263      1647\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3803   3892   5280      6263      1647\n",
      "transfe_506_> MODEL is an average of all templates.\n",
      "transfe_511_> Number of templates for coordinate transfer:        1\n",
      "              After transferring coordinates of the equivalent template atoms,\n",
      "              there are defined, undefined atoms in MODEL:     3323      480\n",
      "openf___224_> Open           GTR1_RAT.ini\n",
      "wrpdb___568_> Residues, atoms, selected atoms:      492     3803     3803\n",
      "make_re_417_> Restraint type to be calculated:  stereo\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3538454    3455.521     3.375\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3546646    3463.521     3.382\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3563030    3479.521     3.398\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3595798    3511.521     3.429\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3628566    3543.521     3.460\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3694102    3607.521     3.523\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3726870    3639.521     3.554\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3759638    3671.521     3.585\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3825174    3735.521     3.648\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      3956246    3863.521     3.773\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      4021782    3927.521     3.835\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      4087318    3991.521     3.898\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      4218390    4119.521     4.023\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      4480534    4375.521     4.273\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      4611606    4503.521     4.398\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      4742678    4631.521     4.523\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      5004822    4887.521     4.773\n",
      "r_stere_606_> Stereochemical restraints were constructed from RTF & PRMF.\n",
      "              Added bond,angle,dihedral,improper restraints  :     3892    5280    5979    1624\n",
      "              Total number of restraints before, now         :        0    16775\n",
      "make_re_422_> Number of previous, current restraints         :        0    16775\n",
      "make_re_423_> Number of previous, current selected restraints:        0    16775\n",
      "make_re_417_> Restraint type to be calculated:  phi-psi_binormal\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      5285187    5161.315     5.040\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch1.bin\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch1.mdt\n",
      "getdata_643_> Protein accepted:  4ZW9\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :    16775    17265\n",
      "make_re_423_> Number of previous, current selected restraints:    16775    17265\n",
      "make_re_417_> Restraint type to be calculated:  omega_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      5285203    5161.331     5.040\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/omega.bin\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/omega.mdt\n",
      "getdata_643_> Protein accepted:  4ZW9\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "omgdel__425_> Unselected all O C +N +CA dihedrals:      514\n",
      "              (This is to avoid clashes between STEREO\n",
      "              and OMEGA_DIHEDRAL restraints)\n",
      "make_re_422_> Number of previous, current restraints         :    17265    17756\n",
      "make_re_423_> Number of previous, current selected restraints:    17265    17242\n",
      "make_re_417_> Restraint type to be calculated:  chi1_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      5285203    5161.331     5.040\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi1.mdt\n",
      "getdata_643_> Protein accepted:  4ZW9\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :    17756    18166\n",
      "make_re_423_> Number of previous, current selected restraints:    17242    17652\n",
      "make_re_417_> Restraint type to be calculated:  chi2_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      5285203    5161.331     5.040\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi2.mdt\n",
      "getdata_643_> Protein accepted:  4ZW9\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      5547347    5417.331     5.290\n",
      "make_re_422_> Number of previous, current restraints         :    18166    18463\n",
      "make_re_423_> Number of previous, current selected restraints:    17652    17949\n",
      "make_re_417_> Restraint type to be calculated:  chi3_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      5547347    5417.331     5.290\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi3.mdt\n",
      "getdata_643_> Protein accepted:  4ZW9\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      6071635    5929.331     5.790\n",
      "make_re_422_> Number of previous, current restraints         :    18463    18585\n",
      "make_re_423_> Number of previous, current selected restraints:    17949    18071\n",
      "make_re_417_> Restraint type to be calculated:  chi4_dihedral\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:      6071635    5929.331     5.790\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi1234.bin\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/chi4.mdt\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "getdata_643_> Protein accepted:  4ZW9\n",
      "getdata_289_> Proteins (all/accepted):        1        1\n",
      "make_re_422_> Number of previous, current restraints         :    18585    18622\n",
      "make_re_423_> Number of previous, current selected restraints:    18071    18108\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "\n",
      "Dynamically allocated memory at amaxhash_contac [B,KiB,MiB]:      7166863    6998.890     6.835\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7429007    7254.890     7.085\n",
      "make_re_422_> Number of previous, current restraints         :    18622    29750\n",
      "make_re_423_> Number of previous, current selected restraints:    18108    29236\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      7953295    7766.890     7.585\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      9001871    8790.890     8.585\n",
      "make_re_422_> Number of previous, current restraints         :    29750    41022\n",
      "make_re_423_> Number of previous, current selected restraints:    29236    40508\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      9526159    9302.890     9.085\n",
      "make_re_422_> Number of previous, current restraints         :    41022    46563\n",
      "make_re_423_> Number of previous, current selected restraints:    40508    46049\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :    46563    49688\n",
      "make_re_423_> Number of previous, current selected restraints:    46049    49174\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :    49688    49688\n",
      "make_re_423_> Number of previous, current selected restraints:    49174    49174\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :    49688    49688\n",
      "make_re_423_> Number of previous, current selected restraints:    49174    49174\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :    49688    49688\n",
      "make_re_423_> Number of previous, current selected restraints:    49174    49174\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :    49688    49688\n",
      "make_re_423_> Number of previous, current selected restraints:    49174    49174\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "make_re_417_> Restraint type to be calculated:  DISTANCE\n",
      "make_re_422_> Number of previous, current restraints         :    49688    49688\n",
      "make_re_423_> Number of previous, current selected restraints:    49174    49174\n",
      "rmdupl__427_>     2942 redundant cosine dihedral restraints were unselected.\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    49688    46232\n",
      "openf___224_> Open           GTR1_RAT.rsr\n",
      "openf___224_> Open           GTR1_RAT.rsr\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:      9733059    9504.940     9.282\n",
      "rdcsr2__307_> Number of restraints read    :    46232\n",
      "              Number of excluded pairs read:        0\n",
      "              Number of pseudo atoms read  :        0\n",
      "rdcsrs__304_> Restraints in memory, selected restraints:    46232    46232\n",
      "              Explicitly excluded atom pairs in memory :        0\n",
      "              Pseudo atoms in memory                   :        0\n",
      "openf___224_> Open           GTR1_RAT.ini\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:      9733091    9504.972     9.282\n",
      "read_mo_297_> Segments, residues, atoms:        1      492     3803\n",
      "read_mo_298_> Segment:        1  1:A  492:A     3803\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:     3803     3803        1        4.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    22158\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    25468\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    27542\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    29986\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    31438\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    32338\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    33466\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    36723\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    38976\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     10116935    9879.819     9.648\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    43653\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    45334\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    46232\n",
      "iupac_m_486_> OE1/2 will be swapped:     -105.3938        2  2\n",
      "iupac_m_487_> NH1/2 swapped:      174.6500       11  11\n",
      "iupac_m_486_> OE1/2 will be swapped:      127.1065       41  41\n",
      "iupac_m_486_> OE1/2 will be swapped:     -163.4520       42  42\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       99.1078       44  44\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:     -176.1842       51  51\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       98.9301       52  52\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5394       89  89\n",
      "iupac_m_487_> NH1/2 swapped:      178.8428       92  92\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      101.1450      112  112\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      103.3597      119  119\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:     -101.6730      194  194\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:      179.1089      223  223\n",
      "iupac_m_487_> NH1/2 swapped:     -173.7269      232  232\n",
      "iupac_m_486_> OE1/2 will be swapped:      132.7451      246  246\n",
      "iupac_m_487_> NH1/2 swapped:      179.0363      249  249\n",
      "iupac_m_487_> NH1/2 swapped:      179.0135      253  253\n",
      "iupac_m_487_> NH1/2 swapped:      167.0130      264  264\n",
      "iupac_m_486_> OE1/2 will be swapped:      166.2803      299  299\n",
      "iupac_m_487_> NH1/2 swapped:      179.4192      333  333\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:     -101.8986      375  375\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       94.6060      416  416\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:      177.7965      458  458\n",
      "iupac_m_485_> OD1/2 will be swapped:      147.7344      461  461\n",
      "iupac_m_487_> NH1/2 swapped:     -177.4171      468  468\n",
      "iupac_m_485_> OD1/2 will be swapped:     -167.5195      476  476\n",
      "iupac_m_486_> OE1/2 will be swapped:      149.5158      480  480\n",
      "iupac_m_486_> OE1/2 will be swapped:      179.5563      481  481\n",
      "openf___224_> Open           ${LIB}/atmcls-mf.lib\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117363    9880.237     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117427    9880.300     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117485    9880.356     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117549    9880.419     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117615    9880.483     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117679    9880.546     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117835    9880.698     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10117963    9880.823     9.649\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10118257    9881.110     9.650\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10118449    9881.298     9.650\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10118953    9881.790     9.650\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10119209    9882.040     9.650\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10120205    9883.013     9.651\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10120589    9883.388     9.652\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10121165    9883.950     9.652\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10123199    9885.937     9.654\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10124095    9886.812     9.655\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10128547    9891.159     9.659\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10129891    9892.472     9.661\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10139509    9901.864     9.670\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10141493    9903.802     9.672\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10162139    9923.964     9.691\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10165147    9926.901     9.694\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10211113    9971.790     9.738\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10215593    9976.165     9.742\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10316813   10075.013     9.839\n",
      "openf___224_> Open           ${LIB}/dist-mf.lib\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10317209   10075.399     9.839\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10318005   10076.177     9.840\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10318801   10076.954     9.841\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10319989   10078.114     9.842\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10320389   10078.505     9.842\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10320989   10079.091     9.843\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10321889   10079.970     9.844\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10323237   10081.286     9.845\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10323437   10081.481     9.845\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10325461   10083.458     9.847\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10325761   10083.751     9.847\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10328797   10086.716     9.850\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10329245   10087.153     9.851\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10333797   10091.599     9.855\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10334469   10092.255     9.856\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10341297   10098.923     9.862\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10342305   10099.907     9.863\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10352549   10109.911     9.873\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10353149   10110.497     9.874\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10353549   10110.888     9.874\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10355061   10112.364     9.875\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10370425   10127.368     9.890\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10371325   10128.247     9.891\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10371925   10128.833     9.891\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10374193   10131.048     9.894\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10397241   10153.556     9.916\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10398585   10154.868     9.917\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10399485   10155.747     9.918\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10402889   10159.071     9.921\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10437461   10192.833     9.954\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10439477   10194.802     9.956\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10440825   10196.118     9.957\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10445929   10201.103     9.962\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10497785   10251.743    10.011\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10500809   10254.696    10.014\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10502833   10256.673    10.016\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10510489   10264.149    10.024\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10588273   10340.110    10.098\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10592809   10344.540    10.102\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10595845   10347.505    10.105\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10607329   10358.720    10.116\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10724005   10472.661    10.227\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10730809   10479.306    10.234\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10735361   10483.751    10.238\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10752589   10500.575    10.254\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10927605   10671.489    10.421\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10937817   10681.462    10.431\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10944645   10688.130    10.438\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     10970485   10713.364    10.462\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11233009   10969.735    10.713\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11248321   10984.688    10.727\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11258565   10994.692    10.737\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11297325   11032.544    10.774\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11691109   11417.099    11.150\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11714077   11439.528    11.171\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11729441   11454.532    11.186\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     11787581   11511.310    11.242\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     12378257   12088.142    11.805\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     12412709   12121.786    11.838\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     12435757   12144.294    11.860\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     12522969   12229.462    11.943\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     13408985   13094.712    12.788\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     13460669   13145.185    12.837\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     13495241   13178.946    12.870\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     13626057   13306.696    12.995\n",
      "\n",
      "Dynamically allocated memory at amaxgroup_restr [B,KiB,MiB]:     14955081   14604.571    14.262\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                                0       0       0       0   12561\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     15339125   14979.614    14.629\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     15521669   15157.880    14.803\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     15886757   15514.411    15.151\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     16616933   16227.474    15.847\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     18077285   17653.598    17.240\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     20997989   20505.848    20.025\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     26839397   26210.348    25.596\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3803    3803\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    46232   46232\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   806222\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -66910.3359\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -66910.335938\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /usr/local/anaconda3/envs/Bioinformatics/lib/modeller-10.1/modlib/surf5.de\n",
      "Pair library             : /usr/local/anaconda3/envs/Bioinformatics/lib/modeller-10.1/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.596001\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.135061\n",
      "Native energy (pair)     :  -392.206012\n",
      "Native energy (surface)  :    -7.276174\n",
      "Native energy (combined) :    -8.944687\n",
      "Z score (pair)           :    -9.678440\n",
      "Z score (surface)        :    -5.260526\n",
      "Z score (combined)       :   -10.612333\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3803    3803\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    46232   46232\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     9167\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        2001.8086\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3892       0      0   0.006   0.006      32.671       1.000\n",
      " 2 Bond angle potential               :    5280       0     12   2.099   2.099      453.48       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2523       0     27  42.138  42.138      690.41       1.000\n",
      " 4 Stereochemical improper torsion pot:    1624       0      0   1.378   1.378      63.642       1.000\n",
      " 5 Soft-sphere overlap restraints     :    9167       0      0   0.001   0.001      2.2678       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   11128       0      0   0.087   0.087      97.086       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11272       0      1   0.115   0.115      165.60       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0     10   4.256   4.256      104.90       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     410       0      4  65.721  65.721      76.413       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     297       0      1  64.189  64.189      86.978       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      1  71.403  71.403      77.857       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0  83.621  83.621      30.237       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5541       0      0   0.308   0.308      102.58       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      22     24  16.574  36.780     -241.14       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3125       0      5   0.711   0.711      258.82       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           GTR1_RAT.V99990001\n",
      "# Heavy relative violation of each residue is written to: GTR1_RAT.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   41853.5625\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13320   1M   2E C   N       7    9   59.24  -69.30  147.68   13.25  -63.60  161.26   19.69\n",
      "    1          2E   2E N   CA      9   10 -144.78  142.50                  -40.30\n",
      "    2  13370  51R  52Y C   N     395  397   85.93   55.90   35.94    2.67  -63.50  162.23   30.30\n",
      "    2         52Y  52Y N   CA    397  398   19.77   39.50                  -43.40\n",
      "    3  13497 178S 179I C   N    1341 1343  -80.58  -63.40   47.70    8.45  -63.40   47.70    8.45\n",
      "    3        179I 179I N   CA   1343 1344  -88.09  -43.60                  -43.60\n",
      "    4  13499 180M 181G C   N    1357 1359  -66.48  -62.40   45.05    6.54  -62.40   45.05    6.54\n",
      "    4        181G 181G N   CA   1359 1360    3.67  -41.20                  -41.20\n",
      "    5  13700 381V 382G C   N    2932 2934 -168.57  -62.40  106.35   19.23  -62.40  106.35   19.23\n",
      "    5        382G 382G N   CA   2934 2935  -47.31  -41.20                  -41.20\n",
      "    6  13791 472A 473S C   N    3648 3650  174.26 -136.60   50.04    1.75  -64.10 -155.50   23.60\n",
      "    6        473S 473S N   CA   3650 3651  160.61  151.20                  -35.00\n",
      "    7  13792 473S 474Q C   N    3654 3656 -138.91 -121.10   56.49    2.91  -63.80  147.03   18.87\n",
      "    7        474Q 474Q N   CA   3656 3657   86.09  139.70                  -40.30\n",
      "    8  13793 474Q 475S C   N    3663 3665  156.96 -136.60   78.24    2.38  -64.10 -168.01   23.82\n",
      "    8        475S 475S N   CA   3665 3666 -167.50  151.20                  -35.00\n",
      "    9  13794 475S 476D C   N    3669 3671  -76.59  -70.90   49.29    2.15  -63.30  121.47   15.83\n",
      "    9        476D 476D N   CA   3671 3672 -160.74  150.30                  -40.00\n",
      "   10  13796 477K 478T C   N    3686 3688  -92.72  -78.10   40.08    2.21  -63.20  157.38   18.38\n",
      "   10        478T 478T N   CA   3688 3689  112.48  149.80                  -42.10\n",
      "   11  13797 478T 479P C   N    3693 3695  -51.01  -58.70   30.47    1.97  -64.50  153.41   12.17\n",
      "   11        479P 479P N   CA   3695 3696  -59.98  -30.50                  147.20\n",
      "   12  13798 479P 480E C   N    3700 3702 -162.81 -117.80   48.94    1.45  -63.60 -168.60   30.97\n",
      "   12        480E 480E N   CA   3702 3703  156.02  136.80                  -40.30\n",
      "   13  13799 480E 481E C   N    3709 3711  -69.79  -69.30   13.72    1.06  -63.60  169.21   22.23\n",
      "   13        481E 481E N   CA   3711 3712  128.79  142.50                  -40.30\n",
      "   14  13800 481E 482L C   N    3718 3720 -106.62 -108.50   24.14    1.32  -63.50  167.87   25.96\n",
      "   14        482L 482L N   CA   3720 3721  156.57  132.50                  -41.20\n",
      "   15  13801 482L 483F C   N    3726 3728 -146.97 -124.20   24.72    0.70  -63.20 -176.95   30.78\n",
      "   15        483F 483F N   CA   3728 3729  152.94  143.30                  -44.30\n",
      "   16  13802 483F 484H C   N    3737 3739 -125.78 -125.60   28.16    1.07  -63.20  165.25   17.77\n",
      "   16        484H 484H N   CA   3739 3740  110.64  138.80                  -42.30\n",
      "   17  13804 485P 486L C   N    3754 3756  -95.45 -108.50   16.39    0.73  -63.50  166.87   21.55\n",
      "   17        486L 486L N   CA   3756 3757  122.58  132.50                  -41.20\n",
      "   18  13805 486L 487G C   N    3762 3764 -105.45  -80.20   38.45    2.65   82.20 -140.08   17.12\n",
      "   18        487G 487G N   CA   3764 3765  145.10  174.10                    8.50\n",
      "   19  13806 487G 488A C   N    3766 3768  -71.64  -68.20    8.05    0.53  -62.50  166.78   27.79\n",
      "   19        488A 488A N   CA   3768 3769  152.58  145.30                  -40.90\n",
      "   20  13807 488A 489D C   N    3771 3773  -88.49  -96.50   12.95    0.53  -63.30  166.29   19.13\n",
      "   20        489D 489D N   CA   3773 3774  124.38  114.20                  -40.00\n",
      "   21  13808 489D 490S C   N    3779 3781  -62.86  -72.40   11.67    0.60  -64.10  179.31   12.92\n",
      "   21        490S 490S N   CA   3781 3782  145.69  152.40                  -35.00\n",
      "   22  13809 490S 491Q C   N    3785 3787  -95.11  -73.00   23.43    1.79  -63.80  176.04   24.35\n",
      "   22        491Q 491Q N   CA   3787 3788  132.93  140.70                  -40.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   12   20  148  326  450  406  568  676  773  733\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "openf___224_> Open           GTR1_RAT.B99990001.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:      492     3803     3803\n",
      "openf___224_> Open           GTR1_RAT.ini\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     26546657   25924.469    25.317\n",
      "read_mo_297_> Segments, residues, atoms:        1      492     3803\n",
      "read_mo_298_> Segment:        1  1:A  492:A     3803\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:     3803     3803        1        4.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    22158\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    25468\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    27542\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    29986\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    31438\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    32338\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    33466\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    36723\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    38976\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    43653\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    45334\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    46232\n",
      "iupac_m_486_> OE1/2 will be swapped:      163.9628        2  2\n",
      "iupac_m_487_> NH1/2 swapped:     -180.0000       11  11\n",
      "iupac_m_487_> NH1/2 swapped:     -178.9051       51  51\n",
      "iupac_m_487_> NH1/2 swapped:     -164.4192       92  92\n",
      "iupac_m_487_> NH1/2 swapped:      178.9730       93  93\n",
      "iupac_m_485_> OD1/2 will be swapped:      175.6912      184  184\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      -93.0268      194  194\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:     -179.3442      218  218\n",
      "iupac_m_487_> NH1/2 swapped:      178.2051      223  223\n",
      "iupac_m_485_> OD1/2 will be swapped:      122.9275      236  236\n",
      "iupac_m_487_> NH1/2 swapped:      179.2705      239  239\n",
      "iupac_m_487_> NH1/2 swapped:      178.3071      269  269\n",
      "iupac_m_487_> NH1/2 swapped:      172.7373      333  333\n",
      "iupac_m_487_> NH1/2 swapped:      179.3406      334  334\n",
      "iupac_m_486_> OE1/2 will be swapped:     -116.6851      426  426\n",
      "iupac_m_487_> NH1/2 swapped:      178.8562      468  468\n",
      "iupac_m_486_> OE1/2 will be swapped:      170.2185      480  480\n",
      "iupac_m_486_> OE1/2 will be swapped:     -177.0065      481  481\n",
      "iupac_m_485_> OD1/2 will be swapped:      170.1385      489  489\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3803    3803\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    46232   46232\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   804521\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -67322.7344\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -67322.734375\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /usr/local/anaconda3/envs/Bioinformatics/lib/modeller-10.1/modlib/surf5.de\n",
      "Pair library             : /usr/local/anaconda3/envs/Bioinformatics/lib/modeller-10.1/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.596001\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.135878\n",
      "Native energy (pair)     :  -425.266298\n",
      "Native energy (surface)  :    -9.173948\n",
      "Native energy (combined) :    -9.035028\n",
      "Z score (pair)           :    -8.968730\n",
      "Z score (surface)        :    -5.480774\n",
      "Z score (combined)       :    -9.356222\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3803    3803\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    46232   46232\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     9187\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1689.6182\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3892       0      0   0.005   0.005      27.534       1.000\n",
      " 2 Bond angle potential               :    5280       0     12   1.942   1.942      395.36       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2523       0     24  41.797  41.797      670.32       1.000\n",
      " 4 Stereochemical improper torsion pot:    1624       0      0   1.165   1.165      47.543       1.000\n",
      " 5 Soft-sphere overlap restraints     :    9187       0      0   0.001   0.001      2.1953       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   11128       0      0   0.078   0.078      77.347       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11272       0      7   0.133   0.133      179.08       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0      6   4.081   4.081      96.444       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     410       0      3  64.123  64.123      40.879       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     297       0      0  62.049  62.049      58.234       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      1  70.225  70.225      78.120       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0 101.956 101.956      23.930       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5541       0      0   0.292   0.292      86.581       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      23     24  15.547  38.187     -253.39       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3125       0      0   0.565   0.565      159.43       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           GTR1_RAT.V99990002\n",
      "# Heavy relative violation of each residue is written to: GTR1_RAT.V99990002\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   38522.8242\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13321   2E   3P C   N      16   18  -67.18  -58.70   10.23    0.87  -64.50  172.00   12.98\n",
      "    1          3P   3P N   CA     18   19  -24.78  -30.50                  147.20\n",
      "    2  13322   3P   4S C   N      23   25  -79.80  -72.40   35.78    1.86  -64.10  138.48   11.04\n",
      "    2          4S   4S N   CA     25   26 -172.59  152.40                  -35.00\n",
      "    3  13370  51R  52Y C   N     395  397   84.09   55.90   33.20    2.53  -63.50  161.42   30.20\n",
      "    3         52Y  52Y N   CA    397  398   21.97   39.50                  -43.40\n",
      "    4  13497 178S 179I C   N    1341 1343  -82.48  -63.40   47.66    8.51  -63.40   47.66    8.51\n",
      "    4        179I 179I N   CA   1343 1344  -87.28  -43.60                  -43.60\n",
      "    5  13700 381V 382G C   N    2932 2934 -168.11  -62.40  105.89   19.16  -62.40  105.89   19.16\n",
      "    5        382G 382G N   CA   2934 2935  -47.39  -41.20                  -41.20\n",
      "    6  13783 464A 465S C   N    3594 3596  -83.81 -136.60   73.72    2.37  -64.10  136.18    8.88\n",
      "    6        465S 465S N   CA   3596 3597   99.74  151.20                  -35.00\n",
      "    7  13784 465S 466G C   N    3600 3602  153.11   82.20  101.45    3.13  -62.40  146.28   26.94\n",
      "    7        466G 466G N   CA   3602 3603  -64.05    8.50                  -41.20\n",
      "    8  13791 472A 473S C   N    3648 3650 -170.85 -136.60   39.73    1.21  -64.10 -172.89   21.26\n",
      "    8        473S 473S N   CA   3650 3651  171.33  151.20                  -35.00\n",
      "    9  13792 473S 474Q C   N    3654 3656 -137.26 -121.10   39.33    2.05  -63.80  161.79   20.89\n",
      "    9        474Q 474Q N   CA   3656 3657  103.85  139.70                  -40.30\n",
      "   10  13793 474Q 475S C   N    3663 3665  176.24 -136.60   50.69    1.61  -64.10 -163.99   22.83\n",
      "   10        475S 475S N   CA   3665 3666  169.76  151.20                  -35.00\n",
      "   11  13794 475S 476D C   N    3669 3671  -95.35  -70.90   66.35    2.29  -63.30  112.67   15.97\n",
      "   11        476D 476D N   CA   3671 3672 -148.02  150.30                  -40.00\n",
      "   12  13795 476D 477K C   N    3677 3679  -60.03  -70.20   19.85    1.63  -62.90  161.78   20.85\n",
      "   12        477K 477K N   CA   3679 3680  157.45  140.40                  -40.80\n",
      "   13  13796 477K 478T C   N    3686 3688  -63.66  -78.10   15.72    1.04  -63.20  161.88   20.79\n",
      "   13        478T 478T N   CA   3688 3689  156.02  149.80                  -42.10\n",
      "   14  13799 480E 481E C   N    3709 3711 -151.84 -117.80   34.04    1.24  -63.60 -162.06   22.97\n",
      "   14        481E 481E N   CA   3711 3712  136.88  136.80                  -40.30\n",
      "   15  13800 481E 482L C   N    3718 3720 -127.36 -108.50   41.57    2.37  -63.50  150.84   18.52\n",
      "   15        482L 482L N   CA   3720 3721   95.45  132.50                  -41.20\n",
      "   16  13801 482L 483F C   N    3726 3728 -170.82 -124.20   48.08    1.54  -63.20 -166.64   33.49\n",
      "   16        483F 483F N   CA   3728 3729  155.06  143.30                  -44.30\n",
      "   17  13802 483F 484H C   N    3737 3739 -134.05 -125.60    8.93    0.37  -63.20 -168.22   20.66\n",
      "   17        484H 484H N   CA   3739 3740  135.91  138.80                  -42.30\n",
      "   18  13803 484H 485P C   N    3747 3749  -52.53  -58.70   11.03    0.67  -64.50  173.57   13.59\n",
      "   18        485P 485P N   CA   3749 3750  -39.64  -30.50                  147.20\n",
      "   19  13804 485P 486L C   N    3754 3756 -114.60 -108.50   12.68    0.61  -63.50 -177.52   28.48\n",
      "   19        486L 486L N   CA   3756 3757  143.62  132.50                  -41.20\n",
      "   20  13805 486L 487G C   N    3762 3764  -84.13  -80.20    5.37    0.38   82.20 -127.86    7.37\n",
      "   20        487G 487G N   CA   3764 3765  170.43  174.10                    8.50\n",
      "   21  13806 487G 488A C   N    3766 3768  -74.57  -68.20    7.16    0.69  -62.50  177.47   29.69\n",
      "   21        488A 488A N   CA   3768 3769  142.05  145.30                  -40.90\n",
      "   22  13807 488A 489D C   N    3771 3773 -128.58  -96.50   32.73    1.33  -63.30  161.49   17.21\n",
      "   22        489D 489D N   CA   3773 3774  107.71  114.20                  -40.00\n",
      "   23  13808 489D 490S C   N    3779 3781  -72.33  -72.40   10.94    0.64  -64.10  176.65   12.33\n",
      "   23        490S 490S N   CA   3781 3782  141.46  152.40                  -35.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   15   24  178  338  426  413  523  678  750  708\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "openf___224_> Open           GTR1_RAT.B99990002.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:      492     3803     3803\n",
      "openf___224_> Open           GTR1_RAT.ini\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     26546657   25924.469    25.317\n",
      "read_mo_297_> Segments, residues, atoms:        1      492     3803\n",
      "read_mo_298_> Segment:        1  1:A  492:A     3803\n",
      "randomi_498_> Atoms,selected atoms,random_seed,amplitude:     3803     3803        1        4.0000\n",
      "randomi_496_> Amplitude is > 0; randomization is done.\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    22158\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    25468\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    27542\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    29986\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    31438\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    32338\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    33466\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    36723\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    38976\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    43653\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    45334\n",
      "pick_re_612_> Number of MODEL atoms, selected restraints;     3803    46232\n",
      "iupac_m_487_> NH1/2 swapped:     -179.8022       11  11\n",
      "iupac_m_486_> OE1/2 will be swapped:     -175.6965       41  41\n",
      "iupac_m_486_> OE1/2 will be swapped:      178.5335       42  42\n",
      "iupac_m_487_> NH1/2 swapped:      177.8160       51  51\n",
      "iupac_m_487_> NH1/2 swapped:     -177.2559       89  89\n",
      "iupac_m_487_> NH1/2 swapped:     -179.0315       93  93\n",
      "iupac_m_485_> OD1/2 will be swapped:      114.0190      177  177\n",
      "iupac_m_487_> NH1/2 swapped:     -179.5114      223  223\n",
      "iupac_m_487_> NH1/2 swapped:     -178.7957      269  269\n",
      "iupac_m_486_> OE1/2 will be swapped:      164.2283      299  299\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       95.0411      320  320\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_487_> NH1/2 swapped:     -176.2885      333  333\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:      -91.5920      395  395\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_483_> CD1/2 CE1/2 swapped:       92.8470      445  445\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_485_> OD1/2 will be swapped:      101.0962      461  461\n",
      "iupac_m_486_> OE1/2 will be swapped:      166.5836      462  462\n",
      "iupac_m_485_> OD1/2 will be swapped:     -177.4742      476  476\n",
      "iupac_m_486_> OE1/2 will be swapped:      134.3751      480  480\n",
      "iupac_m_486_> OE1/2 will be swapped:     -139.6363      481  481\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3803    3803\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    46232   46232\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   807185\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -66834.9688\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -66834.968750\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /usr/local/anaconda3/envs/Bioinformatics/lib/modeller-10.1/modlib/surf5.de\n",
      "Pair library             : /usr/local/anaconda3/envs/Bioinformatics/lib/modeller-10.1/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.596001\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.133916\n",
      "Native energy (pair)     :  -407.560726\n",
      "Native energy (surface)  :    -7.540508\n",
      "Native energy (combined) :    -8.492895\n",
      "Z score (pair)           :    -8.546257\n",
      "Z score (surface)        :    -5.404054\n",
      "Z score (combined)       :    -8.957052\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3803    3803\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    46232   46232\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     9310\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1992.6729\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3892       0      0   0.005   0.005      30.657       1.000\n",
      " 2 Bond angle potential               :    5280       0      9   2.035   2.035      427.31       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2523       0     26  41.996  41.996      678.14       1.000\n",
      " 4 Stereochemical improper torsion pot:    1624       0      0   1.257   1.257      55.009       1.000\n",
      " 5 Soft-sphere overlap restraints     :    9310       0      0   0.002   0.002      4.7055       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   11128       0      0   0.092   0.092      96.656       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11272       3     11   0.155   0.155      243.88       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0      6   4.104   4.104      97.549       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     410       0      1  63.949  63.949      56.385       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     297       0      1  69.282  69.282      72.127       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      0  66.565  66.565      81.853       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0  75.796  75.796      26.714       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5541       0      0   0.303   0.303      96.428       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      28     30  19.498  39.988     -187.41       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3125       0      5   0.649   0.649      212.67       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           GTR1_RAT.V99990003\n",
      "# Heavy relative violation of each residue is written to: GTR1_RAT.V99990003\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   41107.9609\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13320   1M   2E C   N       7    9   75.14  -63.60  166.84   21.23  -63.60  166.84   21.23\n",
      "    1          2E   2E N   CA      9   10 -132.96  -40.30                  -40.30\n",
      "    2  13321   2E   3P C   N      16   18  -70.48  -58.70   37.55    2.31  -64.50  142.16   10.93\n",
      "    2          3P   3P N   CA     18   19    5.16  -30.50                  147.20\n",
      "    3  13322   3P   4S C   N      23   25   73.40   56.90   26.92    1.35  -64.10  146.35   19.58\n",
      "    3          4S   4S N   CA     25   26   15.13   36.40                  -35.00\n",
      "    4  13370  51R  52Y C   N     395  397   86.48   55.90   38.38    2.66  -63.50  161.43   30.07\n",
      "    4         52Y  52Y N   CA    397  398   16.31   39.50                  -43.40\n",
      "    5  13497 178S 179I C   N    1341 1343  -82.02  -63.40   48.83    8.69  -63.40   48.83    8.69\n",
      "    5        179I 179I N   CA   1343 1344  -88.74  -43.60                  -43.60\n",
      "    6  13700 381V 382G C   N    2932 2934 -167.60  -62.40  105.36   19.05  -62.40  105.36   19.05\n",
      "    6        382G 382G N   CA   2934 2935  -47.06  -41.20                  -41.20\n",
      "    7  13774 455T 456K C   N    3522 3524    5.77  -62.90  112.01   13.14  -70.20  118.01    9.85\n",
      "    7        456K 456K N   CA   3524 3525 -129.29  -40.80                  140.40\n",
      "    8  13775 456K 457G C   N    3531 3533  -61.50  -62.40   32.93    4.96   82.20  144.68   10.27\n",
      "    8        457G 457G N   CA   3533 3534   -8.28  -41.20                    8.50\n",
      "    9  13783 464A 465S C   N    3594 3596  -68.23 -136.60   68.53    2.60  -64.10  178.51   13.18\n",
      "    9        465S 465S N   CA   3596 3597  146.53  151.20                  -35.00\n",
      "   10  13784 465S 466G C   N    3600 3602  110.31   82.20   84.58    2.75  -62.40  175.30   29.95\n",
      "   10        466G 466G N   CA   3602 3603  -71.27    8.50                  -41.20\n",
      "   11  13788 469Q 470G C   N    3635 3637 -161.08   78.70  143.11    4.45  -62.40 -174.17   25.28\n",
      "   11        470G 470G N   CA   3637 3638  116.27 -166.10                  -41.20\n",
      "   12  13789 470G 471G C   N    3639 3641  141.06   82.20   78.16    2.64  -62.40  156.54   28.03\n",
      "   12        471G 471G N   CA   3641 3642  -42.93    8.50                  -41.20\n",
      "   13  13791 472A 473S C   N    3648 3650  -69.12  -72.40   12.00    0.60  -64.10  175.94   12.46\n",
      "   13        473S 473S N   CA   3650 3651  140.86  152.40                  -35.00\n",
      "   14  13792 473S 474Q C   N    3654 3656 -129.80 -121.10   21.58    0.85  -63.80  173.31   28.96\n",
      "   14        474Q 474Q N   CA   3656 3657  159.44  139.70                  -40.30\n",
      "   15  13793 474Q 475S C   N    3663 3665  -61.57  -72.40   42.42    2.15  -64.10  146.41   10.76\n",
      "   15        475S 475S N   CA   3665 3666  111.38  152.40                  -35.00\n",
      "   16  13794 475S 476D C   N    3669 3671 -125.95  -96.50   30.37    1.25  -63.30  173.32   18.60\n",
      "   16        476D 476D N   CA   3671 3672  121.60  114.20                  -40.00\n",
      "   17  13795 476D 477K C   N    3677 3679  -82.73  -70.20   27.46    2.24  -62.90  158.01   19.52\n",
      "   17        477K 477K N   CA   3679 3680  115.97  140.40                  -40.80\n",
      "   18  13796 477K 478T C   N    3686 3688 -117.92 -124.80    9.38    0.31  -63.20 -172.61   20.91\n",
      "   18        478T 478T N   CA   3688 3689  137.12  143.50                  -42.10\n",
      "   19  13798 479P 480E C   N    3700 3702 -135.37 -117.80   24.59    1.31  -63.60  175.27   20.44\n",
      "   19        480E 480E N   CA   3702 3703  119.59  136.80                  -40.30\n",
      "   20  13799 480E 481E C   N    3709 3711 -115.33 -117.80   41.09    2.02  -63.60  145.58   17.16\n",
      "   20        481E 481E N   CA   3711 3712   95.78  136.80                  -40.30\n",
      "   21  13800 481E 482L C   N    3718 3720 -117.43 -108.50   76.12    4.20  -63.50  111.95   13.71\n",
      "   21        482L 482L N   CA   3720 3721   56.91  132.50                  -41.20\n",
      "   22  13801 482L 483F C   N    3726 3728  178.42 -124.20   77.04    2.03  -63.20  169.27   30.23\n",
      "   22        483F 483F N   CA   3728 3729 -165.29  143.30                  -44.30\n",
      "   23  13802 483F 484H C   N    3737 3739  -85.69  -67.60   18.41    1.48  -63.20 -179.70   21.23\n",
      "   23        484H 484H N   CA   3739 3740  136.59  140.00                  -42.30\n",
      "   24  13804 485P 486L C   N    3754 3756 -106.56 -108.50    7.72    0.39  -63.50  171.72   21.78\n",
      "   24        486L 486L N   CA   3756 3757  125.04  132.50                  -41.20\n",
      "   25  13805 486L 487G C   N    3762 3764   82.35   78.70   11.96    0.20   82.20  174.01    8.46\n",
      "   25        487G 487G N   CA   3764 3765 -177.49 -166.10                    8.50\n",
      "   26  13806 487G 488A C   N    3766 3768  -77.96  -68.20   13.03    0.83  -62.50  165.88   27.97\n",
      "   26        488A 488A N   CA   3768 3769  153.94  145.30                  -40.90\n",
      "   27  13807 488A 489D C   N    3771 3773 -122.73  -96.50   30.31    1.24  -63.30  151.17   16.13\n",
      "   27        489D 489D N   CA   3773 3774   99.00  114.20                  -40.00\n",
      "   28  13809 490S 491Q C   N    3785 3787  -89.20  -73.00   20.95    1.23  -63.80  167.64   26.00\n",
      "   28        491Q 491Q N   CA   3787 3788  153.99  140.70                  -40.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    1   23   23  155  337  452  448  530  645  759  719\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "openf___224_> Open           GTR1_RAT.B99990003.pdb\n",
      "wrpdb___568_> Residues, atoms, selected atoms:      492     3803     3803\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf     DOPE score    GA341 score\n",
      "----------------------------------------------------------------------\n",
      "GTR1_RAT.B99990001.pdb        2001.80859   -66910.33594        1.00000\n",
      "GTR1_RAT.B99990002.pdb        1689.61816   -67322.73438        1.00000\n",
      "GTR1_RAT.B99990003.pdb        1992.67285   -66834.96875        1.00000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from modeller import *\n",
    "from modeller.automodel import *\n",
    "#from modeller import soap_protein_od\n",
    "\n",
    "env = environ()\n",
    "a = automodel(env, alnfile='GTR1_RAT-4ZW9.ali',knowns='4ZW9', sequence='GTR1_RAT',\n",
    "              assess_methods=(assess.DOPE,assess.GA341))\n",
    "a.starting_model = 1\n",
    "a.ending_model = 3 #va a presentar hasta 3 modelos diferentes\n",
    "a.make()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40395332   39448.566    38.524\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40395860   39449.082    38.524\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40444460   39496.543    38.571\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40457760   39509.531    38.584\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/radii14.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "openf___224_> Open           $(LIB)/top_heav.lib\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "openf___224_> Open           $(LIB)/par.lib\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40597012   39645.520    38.716\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40598612   39647.082    38.718\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40602592   39650.969    38.722\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40604592   39652.922    38.724\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40607592   39655.852    38.726\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40612092   39660.246    38.731\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40617664   39665.688    38.736\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40620464   39668.422    38.739\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40624664   39672.523    38.743\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40630964   39678.676    38.749\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40633736   39681.383    38.751\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40635136   39682.750    38.753\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                              227     561     661     112       0\n",
      "openf___224_> Open           GTR1_HUMAN.B99990003.pdb\n",
      "readlinef__W> File: GTR1_HUMAN.B99990003.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40646147   39693.504    38.763\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40646223   39693.578    38.763\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40646394   39693.742    38.763\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40647244   39694.574    38.764\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40648502   39695.805    38.765\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40650406   39697.664    38.767\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40650446   39697.703    38.767\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40650536   39697.789    38.767\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40653392   39700.578    38.770\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40653452   39700.637    38.770\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40653587   39700.770    38.770\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40657871   39704.953    38.774\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40657959   39705.039    38.774\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40658157   39705.230    38.775\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40664583   39711.508    38.781\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40664715   39711.637    38.781\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40665012   39711.926    38.781\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40674634   39721.320    38.790\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40674834   39721.516    38.791\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40675284   39721.957    38.791\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40689734   39736.070    38.805\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40690034   39736.359    38.805\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40690709   39737.020    38.806\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40712367   39758.172    38.826\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40712815   39758.609    38.827\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40713823   39759.594    38.828\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40746327   39791.336    38.859\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40746999   39791.992    38.859\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40748511   39793.469    38.861\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40797267   39841.082    38.907\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     41145107   40180.770    39.239\n",
      "read_mo_297_> Segments, residues, atoms:        1      492     3812\n",
      "read_mo_298_> Segment:        1     1   492     3812\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     41160193   40195.500    39.253\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     41443809   40472.469    39.524\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     41448679   40477.227    39.529\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     41732295   40754.195    39.799\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     41789935   40810.484    39.854\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3811   3901      0         0      1652\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     41953927   40970.633    40.010\n",
      "\n",
      "patch_______> segment topology patched using RTF:     1  ;  MET  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3811   3901   5291      6283      1652\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     42112983   41125.961    40.162\n",
      "patch_______> segment topology patched using RTF:   492  ;  VAL  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3812   3902   5293      6285      1653\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      492   3812   3902   5293      6285      1653\n",
      "transfe_506_> MODEL is an average of all templates.\n",
      "transfe_511_> Number of templates for coordinate transfer:        1\n",
      "              After transferring coordinates of the equivalent template atoms,\n",
      "              there are defined, undefined atoms in MODEL:     3812        0\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "Dynamically allocated memory at amaxhash_contac [B,KiB,MiB]:     28493470   27825.654    27.173\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     28877562   28200.744    27.540\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     28969050   28290.088    27.627\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     29152026   28468.775    27.802\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     29517978   28826.150    28.151\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     30249882   29540.900    28.849\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     31713690   30970.400    30.245\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     34641306   33829.398    33.037\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     40496538   39547.398    38.621\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   809064\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -67534.6328\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 2 Bond angle potential               :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 3 Stereochemical cosine torsion poten:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 4 Stereochemical improper torsion pot:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 5 Soft-sphere overlap restraints     :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "14 Sidechain Chi_1 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "31 Non-bonded restraints              :  809036       0    823   6.872   6.872     -67535.       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "do_prof_492W> Are you sure you want to calculate a normalized energy profile!\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           GTR1_HUMAN_M3.profile\n",
      "# Energy of each residue is written to: GTR1_HUMAN_M3.profile\n",
      "# The profile IS normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:   13\n",
      "# The sum of all numbers in the file:     -21.0379\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -67534.632812\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "-67534.6328125"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Evaluar el modelo con DOPE, se obtienen lso perfiles de pseudo energia\n",
    "from modeller import *\n",
    "from modeller.scripts import complete_pdb\n",
    "\n",
    "log.verbose()    # request verbose output\n",
    "env = environ()\n",
    "env.libs.topology.read(file='$(LIB)/top_heav.lib') # read topology\n",
    "env.libs.parameters.read(file='$(LIB)/par.lib') # read parameters\n",
    "\n",
    "# read model file\n",
    "mdl = complete_pdb(env, 'GTR1_HUMAN.B99990003.pdb')\n",
    "\n",
    "# Assess with DOPE:\n",
    "s = selection(mdl)   # all atom selection\n",
    "s.assess_dope(output='ENERGY_PROFILE NO_REPORT', file='GTR1_HUMAN_M3.profile',\n",
    "              normalize_profile=True, smoothing_window=15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40395332   39448.566    38.524\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40395860   39449.082    38.524\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40444460   39496.543    38.571\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     40457760   39509.531    38.584\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/radii14.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL9v24}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "openf___224_> Open           $(LIB)/top_heav.lib\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "openf___224_> Open           $(LIB)/par.lib\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40597012   39645.520    38.716\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40598612   39647.082    38.718\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40602592   39650.969    38.722\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40604592   39652.922    38.724\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40607592   39655.852    38.726\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40612092   39660.246    38.731\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40617664   39665.688    38.736\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40620464   39668.422    38.739\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40624664   39672.523    38.743\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40630964   39678.676    38.749\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40633736   39681.383    38.751\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     40635136   39682.750    38.753\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                              227     561     661     112       0\n",
      "io_data____W> Setting io.atom_files_directory to a colon-delimited string is\n",
      "              deprecated, as it is not robust on Windows systems. Set it to\n",
      "              a list of directories instead. For example:\n",
      "              env.io.atom_files_directory = ['./', '../atom_files']\n",
      "openf___224_> Open           4ZW9.pdb\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40646147   39693.504    38.763\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40646223   39693.578    38.763\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40646394   39693.742    38.763\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40647244   39694.574    38.764\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40648502   39695.805    38.765\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40650406   39697.664    38.767\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40650446   39697.703    38.767\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40650536   39697.789    38.767\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40653392   39700.578    38.770\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40653452   39700.637    38.770\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40653587   39700.770    38.770\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40657871   39704.953    38.774\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40657959   39705.039    38.774\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40658157   39705.230    38.775\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40664583   39711.508    38.781\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40664715   39711.637    38.781\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40665012   39711.926    38.781\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40674634   39721.320    38.790\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40674834   39721.516    38.791\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40675284   39721.957    38.791\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40689734   39736.070    38.805\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40690034   39736.359    38.805\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40690709   39737.020    38.806\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40712367   39758.172    38.826\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40712815   39758.609    38.827\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40713823   39759.594    38.828\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40746327   39791.336    38.859\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     40746999   39791.992    38.859\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40748511   39793.469    38.861\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     40797267   39841.082    38.907\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     41145107   40180.770    39.239\n",
      "read_mo_297_> Segments, residues, atoms:        1      470     3607\n",
      "read_mo_298_> Segment:        1     1 A   470 A     3607\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     41159555   40194.879    39.253\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     41429315   40458.316    39.510\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     41433987   40462.879    39.515\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     41703747   40726.316    39.772\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     41758179   40779.473    39.824\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      470   3607   3688      0         0      1558\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     41913223   40930.883    39.972\n",
      "\n",
      "patch_______> segment topology patched using RTF:     1  ;  MET  ;  NTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      470   3607   3688   5004      5939      1558\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     42072279   41086.211    40.123\n",
      "patch_______> segment topology patched using RTF:   470  ;  HIS  ;  CTER\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      470   3608   3689   5006      5941      1559\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues  atoms  bonds angles dihedrals impropers:\n",
      "                     1      470   3608   3689   5006      5941      1559\n",
      "transfe_506_> MODEL is an average of all templates.\n",
      "transfe_511_> Number of templates for coordinate transfer:        1\n",
      "              After transferring coordinates of the equivalent template atoms,\n",
      "              there are defined, undefined atoms in MODEL:     3607        1\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  HIS:OXT  HIS\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "Dynamically allocated memory at amaxhash_contac [B,KiB,MiB]:     28422562   27756.408    27.106\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     28801758   28126.717    27.467\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     28888350   28211.279    27.550\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     29061534   28380.404    27.715\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     29407902   28718.654    28.046\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     30100638   29395.154    28.706\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     31486110   30748.154    30.027\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     34257054   33454.156    32.670\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     39798942   38866.156    37.955\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      470\n",
      "Number of all, selected real atoms                :     3608    3608\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   784153\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -70655.3906\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 2 Bond angle potential               :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 3 Stereochemical cosine torsion poten:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 4 Stereochemical improper torsion pot:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 5 Soft-sphere overlap restraints     :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "14 Sidechain Chi_1 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "31 Non-bonded restraints              :  784060       0    338   6.892   6.892     -70655.       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "do_prof_492W> Are you sure you want to calculate a normalized energy profile!\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           4ZW9.profile\n",
      "# Energy of each residue is written to: 4ZW9.profile\n",
      "# The profile IS normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:   13\n",
      "# The sum of all numbers in the file:     -21.5738\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -70655.390625\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "-70655.390625"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Evaluar el templado con DOPE, se obiene el perfil de pseudo energia\n",
    "log.verbose()    # request verbose output\n",
    "env = environ()\n",
    "env.libs.topology.read(file='$(LIB)/top_heav.lib') # read topology\n",
    "env.libs.parameters.read(file='$(LIB)/par.lib') # read parameters\n",
    "\n",
    "# directories for input atom files\n",
    "env.io.atom_files_directory = './:../atom_files'\n",
    "\n",
    "# read model file\n",
    "mdl = complete_pdb(env, '4ZW9.pdb', model_segment=('FIRST:A', 'LAST:A'))\n",
    "\n",
    "s = selection(mdl)\n",
    "s.assess_dope(output='ENERGY_PROFILE NO_REPORT', file='4ZW9.profile',\n",
    "              normalize_profile=True, smoothing_window=15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "environ____W> The class 'environ' is deprecated; use 'Environ' instead\n",
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     26723299   26096.973    25.485\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     26723827   26097.488    25.486\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     26772427   26144.949    25.532\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     26785727   26157.938    25.545\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL10v1}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "alignment__W> The class 'alignment' is deprecated; use 'Alignment' instead\n",
      "openf___224_> Open           GTR1_RAT-6THA.ali\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     26788321   26160.469    25.547\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     26789771   26161.887    25.549\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     26792671   26164.719    25.551\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     26798471   26170.383    25.557\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     26810071   26181.711    25.568\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     27050879   26416.875    25.798\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     27052667   26418.621    25.799\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     27054631   26420.539    25.801\n",
      "\n",
      "Read the alignment from file       : GTR1_RAT-6THA.ali\n",
      "\n",
      "Total number of alignment positions:   492\n",
      "\n",
      "  #  Code        #_Res #_Segm PDB_code    Name\n",
      "-------------------------------------------------------------------------------\n",
      "  1       6THA     448      1    6THA.pdb MOL_ID  1; MOLECULE  SOLUTE CARRIER FAMILY 2, FACILITATED GLUCOSE TRAN MEMBER 1;\n",
      "  2   GTR1_RAT     492      1    GTR1_RAT\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '6THA.profile'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-6471c4b96a30>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     31\u001b[0m \u001b[0ma\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodeller\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0malignment\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'GTR1_RAT-6THA.ali'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     32\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 33\u001b[0;31m \u001b[0mtemplate\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_profile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'6THA.profile'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0ma\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'6THA'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     34\u001b[0m \u001b[0mmodel\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_profile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'GTR1_RAT_M1.profile'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0ma\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'GTR1_RAT'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     35\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-13-6471c4b96a30>\u001b[0m in \u001b[0;36mget_profile\u001b[0;34m(profile_file, seq)\u001b[0m\n\u001b[1;32m     14\u001b[0m        the alignment sequence `seq`.\"\"\"\n\u001b[1;32m     15\u001b[0m     \u001b[0;31m# Read all non-comment and non-blank lines from the file:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m     \u001b[0mf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprofile_file\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"r\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     17\u001b[0m     \u001b[0mvals\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '6THA.profile'"
     ]
    }
   ],
   "source": [
    "import pylab\n",
    "import modeller\n",
    "\n",
    "def r_enumerate(seq):\n",
    "    \"\"\"Enumerate a sequence in reverse order\"\"\"\n",
    "    # Note that we don't use reversed() since Python 2.3 doesn't have it\n",
    "    num = len(seq) - 1\n",
    "    while num >= 0:\n",
    "        yield num, seq[num]\n",
    "        num -= 1\n",
    "\n",
    "def get_profile(profile_file, seq):\n",
    "    \"\"\"Read `profile_file` into a Python array, and add gaps corresponding to\n",
    "       the alignment sequence `seq`.\"\"\"\n",
    "    # Read all non-comment and non-blank lines from the file:\n",
    "    f = open(profile_file, \"r\")\n",
    "    vals = []\n",
    "    for line in f:\n",
    "        if not line.startswith('#') and len(line) > 10:\n",
    "            spl = line.split()\n",
    "            vals.append(float(spl[-1]))\n",
    "    # Insert gaps into the profile corresponding to those in seq:\n",
    "    for n, res in r_enumerate(seq.residues):\n",
    "        for gap in range(res.get_leading_gaps()):\n",
    "            vals.insert(n, None)\n",
    "    # Add a gap at position '0', so that we effectively count from 1:\n",
    "    vals.insert(0, None)\n",
    "    return vals\n",
    "\n",
    "e = modeller.environ()\n",
    "a = modeller.alignment(e, file='GTR1_RAT-6THA.ali')\n",
    "\n",
    "template = get_profile('6THA.profile', a['6THA'])\n",
    "model = get_profile('GTR1_RAT_M1.profile', a['GTR1_RAT'])\n",
    "\n",
    "# Plot the template and model profiles in the same plot for comparison:\n",
    "pylab.figure(1, figsize=(10,6))\n",
    "pylab.xlabel('Alignment position')\n",
    "pylab.ylabel('DOPE per-residue score')\n",
    "pylab.plot(model, color='red', linewidth=2, label='Model')\n",
    "pylab.plot(template, color='green', linewidth=2, label='Template')\n",
    "pylab.legend()\n",
    "pylab.savefig('dope_profile.png', dpi=65)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
